draft-ietf-quic-transport.md

```diff
@@ -2352,171 +2361,136 @@ shown in the following figure and described below.
+                            +--------+
+                            |        |
+                            |  open  |
                             |        |
                             +--------+
+                                 |
+                      send STREAM with FIN flag
+                         send RST_STREAM
+                                 |
+                                 v
+                            +--------+
+                            |        |
+                            | closed |
+                            |        |
+                            +--------+
+~~~
+{: #stream-state-send title="Stream States for Sending"}
 
-   send:   endpoint sends this frame
-   recv:   endpoint receives this frame
 
-   STREAM: a STREAM frame
-   FIN:    FIN flag in a STREAM frame
-   RST:    RST_STREAM frame
-   MSD:    MAX_STREAM_DATA frame
-   SB:     STREAM_BLOCKED frame
-~~~
-{: #stream-lifecycle title="Lifecycle of a stream"}
+### Stream States for Receiving
 
-Note that this diagram shows stream state transitions and the frames and flags
-that affect those transitions only.  It is possible for a single frame to cause
-two transitions: receiving a RST_STREAM frame, or a STREAM frame with the FIN
-flag cause the stream state to move from "idle" to "open" and then immediately
-to one of the "half-closed" states.
+{{stream-state-recv}} illustrates the states and transitions that apply to a
+stream at the receiving endpoint.
 
-The recipient of a frame that changes stream state will have a delayed view of
-the state of a stream while the frame is in transit.  Endpoints do not
-coordinate the creation of streams; they are created unilaterally by either
-endpoint.  Endpoints can use acknowledgments to understand the peer's subjective
-view of stream state at any given time.
+~~~
+                            +--------+
+                            |        |
+                            |  idle  |
+                            |        |
+                            +--------+
+                                 |
+                     receive STREAM, RST_STREAM or
+                           STREAM_BLOCKED
+                     or receive these frames for a
+                         higher-numbered stream
+                                 |
+                                 v
+                            +--------+
+                            |        |
+                            |  open  |
+                            |        |
+                            +--------+
+                                 |
+                     receive STREAM with FIN flag
+                        receive RST_STREAM
+                                 |
+                                 v
+                            +--------+
+                            |        |
+                            | closed |
+                            |        |
+                            +--------+
+~~~
+{: #stream-state-recv title="Stream States for Receipt"}
 
```

Comment:
This new diagram shows an undocumented change in the meaning of RST_STREAM. In the current version, RST_STREAM is bilateral. "After sending a RST_STREAM, an endpoint ceases transmission of STREAM frames on
the identified stream.  A receiver of RST_STREAM can discard any data that it
already received on that stream.  An endpoint sends a RST_STREAM in response to
a RST_STREAM unless the stream is already closed." Obviously, this "send an RST back" will not work in an unilateral setup, and at a minimum that text should be updated. But if we want to keep a "right of refusal", we need a new frame, maybe "GOAWAY_STREAM", to indicate that the receiver is now discarding that stream.